src/napi/node_api_value.c

```diff
@@ -483,3 +484,53 @@ napi_status napi_strict_equals(napi_env env, napi_value lhs, napi_value rhs,
 
   return napi_assign_bool(jerry_get_boolean_value(is_equal), result);
 }
+
+napi_status napi_create_promise(napi_env env, napi_deferred* deferred,
+                                napi_value* promise) {
+  jerry_value_t jpromise = jerry_create_promise();
+  if (jerry_value_is_error(jpromise)) {
+    NAPI_INTERNAL_CALL(napi_throw(env, AS_NAPI_VALUE(jpromise)));
+    NAPI_RETURN(napi_pending_exception);
+  }
+  napi_assign_nvalue(jpromise, promise);
+  *deferred = malloc(sizeof(napi_value*));
```

Comment:
out value in napi_* functions could be null though the case might mean nothing.

Reply:
Do you mean the user can give a non-NULL ptr as the deferred value?

Reply:
The API requests the user to do the following:
create a `napi_deferred deferred;`
then pass its memory address to the `napi_create_promise` function. It then creates the deferred object which will be freed with `napi_resolve/reject_deferred` function.

Reply:
I agree with @legendecas, guard the assignment to prevent null pointer dereference.